modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -174,5 +188,23 @@ def enum_zones
       logger.debug "Enumerated authoritative dns zones: #{zones}"
       zones
     end
+
+    def enum_records(zone_name, node_name, type)
+      records = []
+      response = execute "/EnumRecords #{zone_name} #{node_name}. /Type #{type}", "EnumRecords", true
+      response.each do |line|
+        line.chomp!
+        logger.debug "Extracting record from dnscmd output '#{line}'"
+        matches = /^@\s+(?<ttl>\d+)\s+(?<type>#{type})\s+(?<record>\S+)$/.match(line)
```

Comment:
Oh right.  I'll change it.  I'd assumed it was a style thing rather than performance.  Thanks to the link to the docs.  Google failed me today.

For fun and my own learning, I ran a few small tests...
On my laptop, with ruby 1.9, the existing regex takes < 0.02**ms** to process each line.  Several orders of magnitude faster than calling `dnscmd` I would have thought. Ruby 2.2 is about twice as slow, but switching `#{type}` with `[[:upper:]]` makes it almost another 10 times faster.  Not capturing the `ttl` and the `type`  made it about 25% faster.